src/tokio.rs

```diff
@@ -127,6 +127,21 @@ where
                 socket.set_state(PgWireConnectionState::ReadyForQuery);
             }
         }
+        PgWireConnectionState::CopyInProgress => {
+            // query or query in progress
+            match message {
+                PgWireFrontendMessage::CopyData(copy_data) => {
+                    copy_handler.on_copy_data(socket, copy_data).await?;
+                }
+                PgWireFrontendMessage::CopyDone(copy_done) => {
+                    copy_handler.on_copy_done(socket, copy_done).await?;
```

Comment:
How about calling ` client.set_state(PgWireConnectionState::ReadyForQuery);` so user' implementation does not need to call it.